src/OAuth2/Config/OAuth2Config.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\OAuth2\Config;
+
+/**
+ * OAuth2 config.
+ *
+ * @author Daniel Kiesel <icodr8@gmail.com>
+ */
+final class OAuth2Config implements \Serializable
```

Comment:
I would remove this class. This is only parameters, you can inject them directly.